src/seq/fastq/reader.jl

```diff
@@ -34,10 +44,28 @@ function Base.eltype{S}(::Type{FASTQReader{S}})
     return FASTQSeqRecord{S}
 end
 
-function FASTQReader(input::IO, quality_encoding::QualityEncoding)
-    return FASTQReader{DNASequence}(BufferedInputStream(input), quality_encoding)
+function FASTQReader(input::IO; quality_encoding=:sanger, fill_ambiguous=nothing)
+    return FASTQReader{DNASequence}(BufferedInputStream(input), sym2qualenc(quality_encoding), fill_ambiguous)
 end
 
-function (::Type{FASTQReader{S}}){S<:Sequence}(input::IO, quality_encoding::QualityEncoding)
-    return FASTQReader{S}(BufferedInputStream(input), quality_encoding)
+function (::Type{FASTQReader{S}}){S<:Sequence}(input::IO;
+                                               quality_encoding=:sanger,
+                                               fill_ambiguous=nothing)
+    return FASTQReader{S}(BufferedInputStream(input), sym2qualenc(quality_encoding), fill_ambiguous)
+end
+
+# Map the quality encoding name to a corresponding QualityEncoding object.
+function sym2qualenc(name::Symbol)
+    if name == :sanger
+        return SANGER_QUAL_ENCODING
+    elseif name == :solexa
+        return SOLEXA_QUAL_ENCODING
+    elseif name == :illumina13
+        return ILLUMINA13_QUAL_ENCODING
+    elseif name == :illumina15
+        return ILLUMINA15_QUAL_ENCODING
+    elseif name == :illumina18
+        return ILLUMINA18_QUAL_ENCODING
+    end
+    error("quality encoding ':$(name)' is not supported")
```

Comment:
A const `Dict` is doable, but I prefer functions because it can handle invalid values and is truly immutable (`Dict` is mutable even if it is defined as a constant).